File: 2025_02/leetcode2025-02-15.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    vector<int> findBall(vector<vector<int>>& grid) {
        int m = grid.size(), n = grid[0].size();
        vector<int> res(n, -1);
        for (int j=0; j<n; j++){
            int y = j;
            bool flag = true;
            for (int i=0; i<m; i++){
                if (grid[i][y]==1 && (y<n-1 && grid[i][y+1]==1)){
                    y++;
                }
                else if (grid[i][y]==-1 && (y>0 && grid[i][y-1]==-1)){
                    y--;
                }
                else{
                    flag = false;
                    break;
                }
            }
            if (flag){
                res[j] = y;
            }
        }
        return res;
    }
};

int main(){
    Solution s;
    vector<vector<int>> grid = {{1,1,1,-1,-1},{1,1,1,-1,-1},{-1,-1,-1,1,1},{1,1,1,1,-1},{-1,-1,-1,-1,-1}};
    vector<int> res = s.findBall(grid);
    for (int i : res){
        cout << i << " ";
    }
    cout << endl;
}

File: leetcode2025/2025_02/leetcode2025-02-15.py
from typing import List

class Solution:
    def findBall(self, grid: List[List[int]]) -> List[int]:
        m, n = len(grid), len(grid[0])
        res = [-1]*n
        for j in range(n):
            y = j
            flag = True
            for i in range(m):
                if grid[i][y]==1 and (y<n-1 and grid[i][y+1]==1):
                    y += 1
                elif grid[i][y]==-1 and (y>0 and grid[i][y-1]==-1):
                    y -= 1
                else:
                    flag = False
                    break
            if flag:
                res[j] = y
        return res

if __name__ == "__main__":
    s = Solution()
    grid = [[1,1,1,-1,-1],[1,1,1,-1,-1],[-1,-1,-1,1,1],[1,1,1,1,-1],[-1,-1,-1,-1,-1]]
    print(s.findBall(grid))
